github-actions/check-cherry-pick/entrypoint.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+set -e
+
+# The full name of the repository
+REPO_NAME=$(jq -r ".repository.full_name" "$GITHUB_EVENT_PATH")
+# The PR number being checked
+PR_NUMBER=$(jq -r ".pull_request.number" "$GITHUB_EVENT_PATH")
+# The patch branch for the repository, using the provided TARGET environment variable if defined
+if [[ -z $TARGET ]]; then
+  TARGET_BRANCH=$(git ls-remote --heads https://github.com/$REPO_NAME.git | grep -E 'refs\/heads\/[0-9]+\.[0-9]+\.x' | cut -d '/' -f3 | sort -r | head -n1)
+else
+  TARGET_BRANCH=$TARGET
+fi
+
+# Retrieve the patch file for the pr
+curl -L -s https://github.com/$REPO_NAME/pull/$PR_NUMBER.diff > /tmp/pull.diff;
+
+# Add target repo as remote, then fetch the target branch from the remote
+git remote add target https://github.com/$REPO_NAME
+git fetch target $TARGET_BRANCH
+
+# Checkout the target branch from its remote
+git checkout --track target/$TARGET_BRANCH
```

Comment:
Do we need `--track` here? I thought it's only needed if we push back to the remote.